plaidml2/op/lib/ops.cc

```diff
@@ -2469,21 +2469,36 @@ Value squeeze(const Value& value) {
     throw std::runtime_error("Squeeze expects 2 arguments");
   }
 
-  // Read arguments
+  // argument 0: tensor to be squeezed
   auto I = args[0].as_tensor();
-  auto raw_axis = args[1].as_int();
-
-  // Validate arguments
   auto ndims = I.shape().ndims();
-  auto axis = normalize_axis(raw_axis, ndims, "squeeze");
   std::vector<TensorDim> I_dims(ndims);
   std::vector<TensorIndex> I_idxs;
   std::vector<TensorDim> O_dims;
   std::vector<TensorIndex> O_idxs;
   I.bind_dims(I_dims);
+
+  // argument 1: axes to squeeze upon
+  std::vector<int64_t> raw_axes;
+  if (args[1].is_int()) {
+    raw_axes.push_back(args[1].as_int());
+  } else if (args[1].is_none()) {
+    for (uint64_t i = 0; i < ndims; ++i) {
+      if (I_dims[i].as_int() == 1) {
```

Comment:
This will throw if any of the dims are symbolic; IMO we should instead skip symbolic dimensions. See edsl.h around line 100

Reply:
Or alternatively don't have an "automatically find the size 1 dims" option; it's a bit weird in this context

Reply:
The "automatically find size 1 dims" feature exists everywhere, I think it's probably good to have.